packages/components/form/__specs__/phone_input.spec.tsx

```diff
@@ -0,0 +1,89 @@
+import { useForm } from "react-hook-form";
+import { fireEvent } from "@testing-library/react";
+
+import PhoneInput, { PhoneInputProps } from "../src/phone_field/phone_input";
+import { Form } from "../src/form";
+import { render, UserRenderResult } from "../test_utils";
+
+const BACKSPACE_KEY = {
+  key: "Backspace",
+  code: 8,
+  inputType: "deleteContentBackward",
+};
+
+const SHARED_EVENT_CONFIG = {
+  key: BACKSPACE_KEY.key,
+  charCode: BACKSPACE_KEY.code,
+  keyCode: BACKSPACE_KEY.code,
+  which: BACKSPACE_KEY.code,
+};
+
+describe("<PhoneInput />", () => {
+  type MockFormProps = Partial<PhoneInputProps>;
+
+  const backspace = (field: Element): void => {
+    let { value } = field as HTMLInputElement;
+
+    const downEvent = fireEvent.keyDown(field, SHARED_EVENT_CONFIG);
+
+    if (downEvent) {
+      value = value.slice(0, -1);
+
+      fireEvent.input(field, {
+        target: { value },
+        inputType: BACKSPACE_KEY.inputType,
+        bubbles: true,
+        cancelable: true,
+      });
+    }
+
+    fireEvent.keyUp(field, SHARED_EVENT_CONFIG);
+  };
+
+  const fillInput = (field: Element, value: string): void => {
+    fireEvent.input(field, { target: { value } });
+  };
```

Comment:
I didn't add these as helpers yet as they seemed pretty focused on the functionality being tested in this component. If we want to abstract them now, though, that works for me.